### PR TITLE
MODINVSTOR-626 Update bulk download endpoint to save holdings UUIDs

### PR DESCRIPTION
**Pull request by andrii-novytskyi**
**PURPOSE:**

https://issues.folio.org/browse/MODINVSTOR-626
Expand the bulk/ids endpoint to be able to get holdings uuids.

**APPROACH:**
1. Added query field recordType. If no record type, then INSTANCE type will be used to search UUIDs.
2. Change url from instance-bulk/ids to record-bulk/ids, permission name, and so on.
3. Added test case for HOLDINGS record type.

**TODO:**
Should be merged together with PRs for below modules:
- [x]  UI-inventory - {https://github.com/folio-org/ui-inventory/pull/1242 }
- [x] data-export - {https://github.com/folio-org/mod-data-export/pull/220 }   
- [x] perf-testing - {https://github.com/folio-org/perf-testing/pull/34 }
- [x] ui-agreeements - {https://github.com/folio-org/ui-agreements/pull/687 }
- [x] ui-plugin-create-inventory-record - {https://github.com/folio-org/ui-plugin-create-inventory-records/pull/43 }

### CODE REVIEW

**Review by illia-borysenko — APPROVED** [review]

**Review by Igor-Gorchakov — APPROVED** [review]

File: descriptors/ModuleDescriptor-template.json [hunk 1]
```diff
@@ -488,13 +488,13 @@
...
     {
-      "id": "instance-bulk",
+      "id": "record-bulk",
       "version": "0.1",
       "handlers": [
         {
           "methods": ["GET"],
-          "pathPattern": "/instance-bulk/ids",
-          "permissionsRequired": ["inventory-storage.instance-bulk.ids.get"]
+          "pathPattern": "/record-bulk/ids",
```
Comment (KVupp) [inline descriptors/ModuleDescriptor-template.json:496]: This is a breaking change, once you merge this UI will not be able to download any UUIDs, block this PR and let UI devs make necessary changes and only then merge this PR

File: descriptors/ModuleDescriptor-template.json [hunk 2]
```diff
@@ -488,13 +488,13 @@
       ]
     },
     {
-      "id": "instance-bulk",
+      "id": "record-bulk",
```
Comment (KVupp) [inline descriptors/ModuleDescriptor-template.json:491]: I am not sure if renaming an interface was done previously in FOLIO, but we need to increment the version as well

Comment (KVupp) [reply descriptors/ModuleDescriptor-template.json:491]: Also verify if this interface or API is used anywhere else in FOLIO other than the ui-inventory, we need to make those changes in that modules as well

File: ramls/record-bulk.raml [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
+title: Inventory Storage Record Bulk API
 version: v0.1
 protocols: [ HTTP, HTTPS ]
 baseUri: http://localhost
 
 documentation:
-  - title: Inventory Storage Instance Bulk Download API
-    content: <b>API for downloading a bulk set of Instance IDs</b>
+  - title: Inventory Storage records Bulk Download API
+    content: <b>API for downloading a bulk set of records IDs</b>
```
Comment (KVupp) [inline ramls/record-bulk.raml:9]: just for readability sake, please add what records can be downloaded and that ID is infact the UUID of the records

File: ramls/record-bulk.raml [hunk 4]
```diff
@@ -41,3 +41,9 @@ resourceTypes:
           enum: [id]
           default: id
           required: false
+        recordType:
+          description: Determines which record type should be used to query the field
```
Comment (KVupp) [inline ramls/record-bulk.raml:45]:
```suggestion
          description: Determines which record type UUIDs should be fetched
```

File: ramls/record-bulk.raml [hunk 5]
```diff
@@ -21,15 +21,15 @@ traits:
...
+    displayName: Records ID Bulk Download API
     type:
       collection-get:
         schemaCollection: ids
         exampleCollection: !include examples/ids.json
     get:
-      description: Retrieve a list of Instance IDs.
+      description: Retrieve a list of record IDs.
       is: [
         searchable: {description: "with valid searchable fields", example: "name=aaa"},
```
[the remark this answers is not in the document]
Comment (andrii-novytskyi) [reply ramls/record-bulk.raml:34]: yes, if you will take a look at the test, you will see it

File: descriptors/ModuleDescriptor-template.json (hunk 1 above)
Comment (andrii-novytskyi) [reply descriptors/ModuleDescriptor-template.json:496]: @KVupp yes, I described it in notes

File: descriptors/ModuleDescriptor-template.json (hunk 2 above)
Comment (andrii-novytskyi) [reply descriptors/ModuleDescriptor-template.json:491]: UI-inventory, data-export and perf-testing are using this interface

Comment (KVupp) [reply descriptors/ModuleDescriptor-template.json:491]: Thanks! Let's add this list as TODO and when PRs are ready tag them here

File: ramls/record-bulk.raml [hunk 6]
```diff
@@ -1,12 +1,12 @@
 #%RAML 1.0
-title: Inventory Storage Instance Bulk API
+title: Inventory Storage Record Bulk API
```
Comment (KVupp) [inline ramls/record-bulk.raml:2]: change version here as well

Comment (andrii-novytskyi) [reply ramls/record-bulk.raml:2]: @KVupp updated

File: descriptors/ModuleDescriptor-template.json (hunk 2 above)
Comment (andrii-novytskyi) [reply descriptors/ModuleDescriptor-template.json:491]: updated

File: ramls/record-bulk.raml (hunk 6 above)
Comment (KVupp) [reply ramls/record-bulk.raml:2]: and correct the base URI as well please

File: ramls/record-bulk.raml [hunk 7]
```diff
@@ -41,3 +41,9 @@ resourceTypes:
           enum: [id]
           default: id
           required: false
+        recordType:
+          description: Determines which record type UUIDs should be fetched
+          type: string
+          enum: [INSTANCE, HOLDINGS]
```
Comment (KVupp) [inline ramls/record-bulk.raml:47]: A minor one but we have singular for instance but plural for holdings, would be easier to make it consistent

File: ramls/record-bulk.raml (hunk 6 above)
Comment (andrii-novytskyi) [reply ramls/record-bulk.raml:2]: done

**Review by KVupp — APPROVED** [review]

File: ramls/record-bulk.raml (hunk 7 above)
Comment (andrii-novytskyi) [reply ramls/record-bulk.raml:47]: for some reason everywhere in folio holdings are used instead of holding, but ok, I will change to holding

Comment (andrii-novytskyi) [reply ramls/record-bulk.raml]: @KVupp  done

File: descriptors/ModuleDescriptor-template.json [hunk 8]
```diff
@@ -488,13 +488,13 @@
       ]
     },
     {
-      "id": "instance-bulk",
-      "version": "0.1",
+      "id": "record-bulk",
```
Comment (marcjohnson-kint) [inline descriptors/ModuleDescriptor-template.json:491]: Given that this interface is being renamed, please could it be given a name that provides more context than only the term `record`.

What records are involved? If it is any inventory record, could `inventory-record-bulk` be an acceptable name?

This name takes up a very general space and could easily conflict with other interface names that are not related to inventory at all.

Comment (andrii-novytskyi) [reply descriptors/ModuleDescriptor-template.json]: make sense, updated, thanks!

**Pull request merged**